src/gui/library.cpp

```diff
@@ -17,10 +19,12 @@ Library::Library(stringw name, unsigned int buffer_size)
 
     for (unsigned int i = 0; i < file_list->getFileCount(); i++)
     {
-        IXMLReader* xml_reader = file_system->createXMLReader(
-            file_list->getFullFileName(i));
+        AutoDrop<IXMLReader> xml_reader(file_system->createXMLReader(
+            file_list->getFullFileName(i)));
+
+        // Make sure we don't leak element.
+        std::auto_ptr<Element> element(new Element(xml_reader.get()));
```

Comment:
Well yeah, it's deprecated as of C++11, but I thought you were restricting yourself to C++03, where it's the only smart pointer in the standard library. If that information is outdated as you said, then there's indeed no reason to not use ``unique_ptr`` instead.